engine/gamesys/src/gamesys/scripts/script_camera.cpp

```diff
@@ -52,15 +52,14 @@ namespace dmGameSystem
     static int Camera_AcquireFocus(lua_State* L)
     {
         DM_LUA_STACK_CHECK(L, 0);
-        dmGameObject::HInstance instance = CheckGoInstance(L);
+        (void)CheckGoInstance(L); // left to check that it's not called from incorrect context.
 
         dmMessage::URL receiver;
         dmMessage::URL sender;
         dmScript::ResolveURL(L, 1, &receiver, &sender);
 
         dmGamesysDDF::AcquireCameraFocus msg;
-        dmMessage::Post(&sender, &receiver, dmGamesysDDF::AcquireCameraFocus::m_DDFDescriptor->m_NameHash, (uintptr_t)instance,
-            0, (uintptr_t)dmGamesysDDF::AcquireCameraFocus::m_DDFDescriptor, &msg, sizeof(msg), 0);
+        dmMessage::Post(&sender, &receiver, dmGamesysDDF::AcquireCameraFocus::m_DDFDescriptor->m_NameHash, 0, 0, (uintptr_t)dmGamesysDDF::AcquireCameraFocus::m_DDFDescriptor, &msg, sizeof(msg), 0);
```

Comment:
In these cases, the "m_UserData1" was never used on the receiving end, thus safe to remove.

Reply:
As I understand for the cases you removed m_UserData1 this logic will never be touched? https://github.com/defold/defold/blob/7cf88519e88c23e0fa725b263db0c23f31a29ef9/engine/gameobject/src/gameobject/gameobject.cpp#L2227-L2238

Reply:
Good catch.
While it's an optimization, I think it's efficient enough to use the GetInstanceFromIdentifier() (a hash table lookup).
This keeps the code cleaner imho, and less "extra knowledge" you need to use these functions.

I'll remove it for now.